### ISSUE TITLE
Unable to fetch variables

### ISSUE THREAD

**Issue by xdnzh** [jshinonome/vscode-q#54]
Hello, 

recently (maybe since 2 weeks ago?) I just found that I kept getting the error notification saying "Unable to fetch variables from [my q port number]". Besides that, querying and everything else seem working just fine. I'm wondering if this is a known issue or maybe I'm missing something? 

I don't have any major changes in q (company managed) or vscode so far as I know. 

Is there any further information that I can provide to better diagnose this problem?

**Comment by jshinonome** [comment]
Hi, could you try to debug with this function?
https://github.com/jshinonome/vscode-q/blob/master/assets/source/query-server-variables.q
I did make a change to skip partitioned tables two weeks ago, because they take too much time.

**Comment by xdnzh** [comment]
Thanks! This makes sense, as we have a lot of partitioned tables for sure. Somehow this issue doesn't occur with kdb 4.0. I will try to debug and come back with update, and thanks for your reply!

**Comment by jshinonome** [comment]
Happy Chinese New Year!! and let me know if there is any bug.

**Comment by xdnzh** [comment]
Happy Chinese New Year! 

With one of our environments (kdb+ 3.1), I got this error:
```
Value error (.Q.pt undefined)
```
by running the function until 
```
r: update b: {.Q.s value x} each n from r where not t within 100 111h, not n in .Q.pt;
```

With another environment (kdb+ 4.0), fetching variables is just fine. 

Is this because my kdb+ 3.1 doesn't have .Q.pt?

**Comment by jshinonome** [comment]
Thanks for the comments. Looks like 3.1 doesn't have .Q.pt, but I don't have 3.1 to test. In this case, I cannot skip partitioned tables in earlier version... I think I will trap this.

**Comment by xdnzh** [comment]
Thanks! Trapping this would definitely help a lot users like me who have to use older version kdb. And again thanks for your excellent work making this plug-in!

**Comment by jshinonome** [comment]
I will fix this in the next release.

**Comment by jshinonome** [comment]
Just released a fix for this, v2.4.6
